### DISCUSSION TITLE
Counting objects using a Webcam

### DISCUSSION THREAD

**Discussion by zaydanr** [ultralytics/ultralytics#9737] [Q&A]
Hi,
I am trying to count the number of toy cars present in a frame. They are all under one class 'cars'. How can I edit this code to export the number of cars to a single variable named carCount? I am using a webcam and yolov8 with the code attached.
[cameraTest.py.zip](https://github.com/ultralytics/ultralytics/files/14885257/cameraTest.py.zip)

**Comment by pderrenger** [comment]
Hello there! 🚗 To count the number of toy cars in each frame and export it to a single variable named `carCount`, you can modify your detection loop in the script. Assuming you're using our typical prediction syntax, here's a quick example of how you could modify your code:

```python
from ultralytics import YOLO

# Load your YOLO model
model = YOLO('yolov8n.pt')

# Assuming you have initialized your webcam into the variable 'cam'
while True:
    # Read frame from webcam
    success, frame = cam.read()
    if not success:
        break

    # Make prediction
    results = model(frame)

    # Count only detections that are cars
    carCount = (results.boxes.cls == 'cars').sum().item()

    print(f"Number of cars in the current frame: {carCount}")
    # You can then use carCount as needed in the rest of your code

# Don't forget to release your camera resource at the end
cam.release()
```

This code fetches frames from your webcam, performs detections, and uses the detection results to count instances of the class `'cars'`. Make sure you replace `'yolov8n.pt'` with the appropriate model path if you're using a custom model, and adjust the class name `'cars'` if your class has a different name. Happy coding! 😊

**Reply by zaydanr** [reply]
Hi, 
On the line carCount = (results.boxes.cls == 'cars').sum().item()

I am receiving this error:
Exception has occurred: AttributeError
'list' object has no attribute 'boxes'
  File "/Users/zaydanr/Downloads/EDDTrafficLightCode/cameraCount.py", line 24, in <module>
    carCount = (results.boxes.cls == 'cars').sum().item()
AttributeError: 'list' object has no attribute 'boxes'

Any idea on what to do next?

**Reply by zaydanr** [reply]
If it makes it easier, the algorithm is ONLY detecting cars, so maybe counting the number of detections would be easier? I have no idea, though.

**Reply by pderrenger** [reply]
Hello! 😊 It looks like you encountered an issue because `model` predictions return a list of `Results` objects, one for each image in your batch. Since you're working with a single frame at a time, you'll want to access the first item in that list. Here's a simple adjustment to count the detections when you're only detecting 'cars':

```python
carCount = len(results[0].boxes)
print(f"Number of cars in the current frame: {carCount}")
```

Since your model is only detecting cars, counting the total number of detections does indeed simplify things. This code snippet takes the number of detected objects in the first (and only) frame's results and prints it out. Happy detecting!
